src/routers/api.js

```diff
@@ -176,101 +181,157 @@ router.post(
-				}
-				if(result.data.claimedOnboardingLld !== false) {
-					return reject('Person already claimed onboarding LLD');
-				}
-				centralizedId = result.data.id;
-				return resolve(centralizedId)
-			}).catch(e =>{
-				return reject('Technical error when checking if fully approved e-resident who didnt claim LLD yet, please let the devs know');
-			})
-
-		})
-
-		centralizedCheckPromises.push(checkEligibleForPersonIdPromise)
+		centralizedAPI.defaults.headers.common["X-token"] = userToken;
+
+		const centralizedCheckPromises = [];
+
+		const checkEligibleForPersonIdPromise = new Promise(
+			(resolve, reject) => {
+				centralizedAPI
+					.get("/e-residents/approved/me")
+					.then((result) => {
+						if (result.status !== 200) {
+							return reject(
+								"You are not a fully approved e-resident or citizen, therefore are not eligible to claim onboarding LLDs"
+							);
+						}
+						if (result.data.claimedOnboardingLld !== false) {
+							return reject(
+								"Person already claimed onboarding LLD"
+							);
+						}
+						centralizedId = result.data.id;
+						return resolve(centralizedId);
+					})
+					.catch((e) =>
+						reject(
+							"Technical error when checking if fully approved e-resident who didnt claim LLD yet, please let the devs know"
+						)
+					);
+			}
+		);
 
-		let checkWalletAddressPromise = new Promise((resolve, reject) => {
-			centralizedAPI.get('/users/me').then((result) => {
-				centralizedWalletAddress = result.data.blockchainAddress;
-				if(centralizedWalletAddress !== usingWalletAddress) {
-					return reject('Wallet address provided and wallet address registered in profile do not match');
-				}
-				return resolve()
-			}).catch(e => {
-				return reject('Technical error when checking if used and registered wallet address match, please let the devs know');
-			});
-		})
+		centralizedCheckPromises.push(checkEligibleForPersonIdPromise);
+
+		const checkWalletAddressPromise = new Promise((resolve, reject) => {
+			centralizedAPI
+				.get("/users/me")
+				.then((result) => {
+					centralizedWalletAddress = result.data.blockchainAddress;
+					if (centralizedWalletAddress !== usingWalletAddress) {
+						return reject(
+							"Wallet address provided and wallet address registered in profile do not match"
+						);
+					}
+					return resolve();
+				})
+				.catch((e) =>
+					reject(
+						"Technical error when checking if used and registered wallet address match, please let the devs know"
+					)
+				);
+		});
 
-		centralizedCheckPromises.push(checkWalletAddressPromise)
+		centralizedCheckPromises.push(checkWalletAddressPromise);
+
+		const checkIfAlreadyHaveLLDPromise = new Promise((resolve, reject) => {
+			api.derive.balances
+				.all(usingWalletAddress)
+				.then((result) => {
+					const availableBalance = new BN(result.availableBalance);
+					// if (!availableBalance.isZero()) {
+					// 	return reject(
```

Comment:
Please dont comment out the business logic.

Redo this, but with only relevant changes, no linter pls